rs/replicated_state/src/metadata_state/tests.rs

```diff
@@ -1571,6 +1571,67 @@ fn compatibility_for_reject_reason() {
     );
 }
 
+#[test]
+fn stream_responses_tracking() {
+    let mut stream = Stream::new(StreamIndexedQueue::with_begin(0.into()), 0.into());
+    let mut expected_counts = BTreeMap::new();
+    assert_eq!(stream.guaranteed_responses_counts(), &expected_counts);
+
+    let response = ResponseBuilder::default()
+        .respondent(*LOCAL_CANISTER)
+        .originator(*REMOTE_CANISTER)
+        .build();
+    stream.push(response.into());
+    expected_counts.insert(*LOCAL_CANISTER, 1);
+    assert_eq!(stream.guaranteed_responses_counts(), &expected_counts);
+
+    // Best-effort responses don't count.
+    let response = ResponseBuilder::default()
+        .respondent(*LOCAL_CANISTER)
+        .originator(*REMOTE_CANISTER)
+        .deadline(CoarseTime::from_secs_since_unix_epoch(1))
+        .build();
+    stream.push(response.into());
+    assert_eq!(stream.guaranteed_responses_counts(), &expected_counts);
+
+    // Response from a different respondent.
+    let response = ResponseBuilder::default()
+        .respondent(*LOCAL_CANISTER)
+        .originator(*REMOTE_CANISTER)
+        .build();
+    stream.push(response.into());
+    *expected_counts.get_mut(&*LOCAL_CANISTER).unwrap() += 1;
```

Comment:
```suggestion
    *expected_counts.get_mut(&LOCAL_CANISTER).unwrap() += 1;
```
Appears to be working without the `*`.